tests/api_workflow/test_api_workflow_compute_worker.py

```diff
@@ -282,14 +300,18 @@ def test_selection_config_from_dict() -> None:
 
 def test_selection_config_from_dict__missing_strategies() -> None:
     cfg = {}
-    selection_cfg = api_workflow_compute_worker.selection_config_from_dict(cfg)
-    assert selection_cfg.strategies == []
+    with pytest.raises(
+        ValidationError,
+        match=r"strategies\n  ensure this value has at least 1 items",
+    ):
+        api_workflow_compute_worker.selection_config_from_dict(cfg)
 
 
 def test_selection_config_from_dict__extra_key() -> None:
     cfg = {"strategies": [], "invalid-key": 0}
     with pytest.raises(
-        TypeError, match="got an unexpected keyword argument 'invalid-key'"
+        ValidationError,
```

Comment:
I think it's okay to keep it as is since probably no one catches this `TypeError`. There is little one can do after catching this error.

A separate topic: handling validation errors properly. Ideally, `ValidationError` should be handled by us and reraised because it's from pydantic and sometimes the error message is purely internal.

Reply:
Is there a good way to handle the pydantic errors? Because ideally we can show the users exactly where the error happened.

Reply:
I mean normally it's okay to just leave pydantic errors as is. It's a problem only when users try to catch the pydantic error types because that means they need to import pydantic in their code as well. However, I don't think that's a big issue because no one probably catches this kind of error.
The bigger issue is about errors irrelevant to user input. Users won't understand what went wrong unless they look into the source code. This can be prevented by proper tests, so let's not be too concerned about this now.